### ISSUE TITLE
Add support for hexagonal ducts with filleted corners

### ISSUE THREAD

**Issue by dlangewisch** [terrapower/armi#1301]
ARMI currently supports Hexagonal ducts with sharp corners, but in many practical cases the hex ducts have rounded (filleted) corners with a specified corner radius, which reduces both the flow area and wetted perimeter of these ducts and increases the hydraulic diameter by as much as ~2.4% (neglecting pins). Capability should be added to support either specifying a corner radius when defining a component of shape `Hexagon`, or perhaps a new `FilletedHexagon` shape should be added with that capability. 

If $D$ denotes the flat-to-flat inner duct diameter (i.e., the value returned by `duct.getDimension('ip')`) and $r$ denotes the corner radius ($0 \le r \le D/2$), then the perimeter, $P$, of the inner duct surface and the cross-sectional area $A$ inside the duct are given, respectively, by:

$$P(D,r) = \underbrace{2\sqrt{3}D}_{P(D,r=0)} \left[1 - \left(1 - \frac{\pi}{2\sqrt{3}}\right)\left(\frac{2r}{D}\right)\right]$$

and

$$A(D,r) = \underbrace{\frac{\sqrt{3}}{2}D^2}_{A(D,r=0)} \left[1 - \left(1 - \frac{\pi}{2\sqrt{3}}\right)\left(\frac{2r}{D}\right)^2 \right]$$

where $P(D, r=0)$ and $A(D, r=0)$ denote the perimeter and cross-sectional area assuming unrounded corners. Note that in the limiting case where $r = D/2$, the hexagon reduces to a circle and we have $A(D, D/2) = \pi D^2 /4$ and $P(D, D/2) = \pi D$ as expected. 

The hydraulic diameter of a filleted duct is given by

$$D_{hyd} = D \left[ \frac{1 - \left(1 - \frac{\pi}{2\sqrt{3}} \right) \left( \frac{2r}{D} \right)^2}{ 1 - \left(1 - \frac{\pi}{2\sqrt{3}} \right) \left( \frac{2r}{D} \right) } \right]$$

The quantity in brackets is maximum when $r \approx 0.256 D$, in which case $D_{hyd} \approx 1.0224 D$

**Comment by andfranklin** [comment]
Adding a note to also verify that correct calculations are being made for wetted perimeter, flow area and hydraulic diameter for unpinned hex blocks with filleted corners.

**Comment by andfranklin** [comment]
Yup, happy to take this one. I'll think I'll be able to take care of it sometime in September.

**Comment by john-science** [comment]
@dlangewisch How do you expect the "wetted perimeter" calculation to differ for this `FilletedHexagon` versus a regular `Hexagon`?

https://github.com/terrapower/armi/blob/1693f1f2ddb0867a43e38cce0d91673cad67126a/armi/reactor/blocks.py#L2314-L2315
